docs/user-guide/plugins.md

```diff
@@ -0,0 +1,193 @@
+# Plugins
+
+## Overview
+
+This guide demonstrates how to write plugins for the
+`argocd` CLI tool. Plugins are a mean to extend `argocd` CLI with new sub-commands,
+allowing for new and custom features which are not included in the main distribution
+of `argocd` CLI.
+
+If you are interested to take a look at the proposal, head over to this [enhancement proposal](https://github.com/argoproj/argo-cd/pull/19624).
```

Comment:
Can you better phrase this? Why someone would want to look at the proposal? 
Maybe something like: To see more in-depth details about how the Argo CD plugin feature works internally, take a look at the enhancement proposal document. Also.. provide the link to the markdown file instead of the PR. It is better to get that merged first.

Reply:
the enhancement proposal doesn't contain information about the internal working of plugin interface. Infact, the proposal revolves more around requesting for an enhancement and why is it needed than how it should be implemented.

Reply:
cc @christianh814

Reply:
after talking wih Leo, we need not to update the docs anymore but only merge it.

Reply:
fixed.